otter/cloud_client.py

```diff
@@ -507,6 +524,25 @@ class NovaComputeFaultError(Exception):
     """
 
 
+@attributes([])
```

Comment:
Its frustrating that default exception doesn't have equality check. The above `_BaseExceptionWithMessage` is a good idea except that it can be renamed to something like `ExceptionWithEquality`?